src/eth_rpc/servers/alchemy_rpc.rs

```diff
@@ -1,41 +1,83 @@
 use futures::future::join_all;
 use jsonrpsee::core::{async_trait, RpcResult};
-use reth_primitives::{Address, BlockId, BlockNumberOrTag};
+use reth_primitives::{Address, BlockId, BlockNumberOrTag, U256};
 
 use crate::eth_provider::contracts::erc20::EthereumErc20;
 use crate::eth_provider::error::EthApiError;
+use crate::eth_provider::provider::EthereumProvider;
 use crate::eth_rpc::api::alchemy_api::AlchemyApiServer;
-use crate::models::balance::TokenBalanceFuture;
-use crate::{eth_provider::provider::EthereumProvider, models::balance::TokenBalances};
+use crate::models::token::{TokenBalance, TokenBalances, TokenMetadata};
 
 /// The RPC module for the Ethereum protocol required by Kakarot.
 #[derive(Debug)]
 pub struct AlchemyRpc<P: EthereumProvider> {
+    /// The provider for interacting with the Ethereum network.
     eth_provider: P,
 }
 
 impl<P: EthereumProvider> AlchemyRpc<P> {
+    /// Creates a new instance of [`AlchemyRpc`].
     pub const fn new(eth_provider: P) -> Self {
         Self { eth_provider }
     }
 }
 
 #[async_trait]
 impl<P: EthereumProvider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
+    /// Retrieves the token balances for a given address.
     #[tracing::instrument(skip_all, ret, fields(address = %address, token_addresses = ?token_addresses))]
```

Comment:
I think we can update the macro (sorry this falls on you lol) and basically I think I would not display the `token_addresses`. I think this will cause a LOT of logs which are props not useful. Same I would say with `ret`. We could have `err` though, what do you think?

Reply:
yes agreed just fixed this